src/logger.c

```diff
@@ -103,7 +103,10 @@ static void log_file_init(struct log_file *f, char *name, int size)
 	    O_WRONLY | O_CREAT | O_APPEND, S_IRUSR | S_IWUSR)) < 0)
 		pexit("open: %s", path_expand(name));
 
-	f->ptr = f->buffer = mem_alloc(size + LINE_BUFFER_SIZE);
+	/* plain will now always be < LINE_BUFFER_SIZE. We add some extra bytes
+	 * so that there is ALWAYS enough buffer to write our line, and we no
+	 * longer have to check length before a write (.pot or .log file) */
+	f->ptr = f->buffer = mem_alloc(size + LINE_BUFFER_SIZE + 512);
```

Comment:
The 512 was way more than needed.  It allows all these 'checks' to be ignored (sep char, password, etc). It is not used in ANY computation, it is just extra waste bytes allocated.

Doing that allowed quite a lot of worthless bullet-proof checking to simply go away.